### PR TITLE
fix: setAccount_balance

### PR DESCRIPTION
**Pull request by willemneal**
This fixes how setAccount_balance so that attached balance isn't doubled added to it.

Fixes issues with passing large u128 when setting balances, etc, by passing a string.  Also context wasn't being reset for the mock vm.

Adds a feature to roll back the memory of the unit test's Wasm memory.  This will help ensure that there is tests are idempotent.

### CODE REVIEW

File: sdk/assembly/__tests__/runtime.spec.ts [hunk 1]
```diff
@@ -770,12 +785,12 @@ describe("promise batches", () => {
       .delete_account("bene.account");
   });
 
-  it("should support contract batch transactions", () => {
-    VMContext.setPrepaid_gas(10000000000000);
-    expect(Context.accountBalance).toBe(u128.from(9));
-    ContractPromiseBatch.create("alice").transfer(u128.from(1));
-    expect(Context.accountBalance).toBe(u128.from(8)); // this number is surprising
-  });
+  // it("should support contract batch transactions", () => {
```
Comment (vgrichina) [inline sdk/assembly/__tests__/runtime.spec.ts:788]: @willemneal why are this and other tests commented out?

File: near-mock-vm/assembly/context.ts [hunk 2]
```diff
@@ -32,17 +32,17 @@ declare function _setRandom_seed(s: string): void;
 /** @internal */
 //@ts-ignore
 @external("vm", "setAttached_deposit")
-declare function _setAttached_deposit(lo: u64, hi: u64): void;
+declare function _setAttached_deposit(_u128: usize): void;
```
Comment (vgrichina) [inline near-mock-vm/assembly/context.ts:35]: @willemneal not sure how does this change work. Does it break backwards compatibility? Or old signature was misleading?

File: sdk/assembly/__tests__/runtime.spec.ts (hunk 1 above)
Comment (willemneal) [reply sdk/assembly/__tests__/runtime.spec.ts:788]: They should be deleted. They are cross contract tests that don't really test anything and should be in unit tests anyway.

File: near-mock-vm/assembly/context.ts (hunk 2 above)
Comment (willemneal) [reply near-mock-vm/assembly/context.ts:35]: It was an internal API. It was my attempt to pass a u128 with the high and low parts a d I went with a string instead. And this usize is the ptr to a utf8 string.

**Review by vgrichina — APPROVED** [review]

**Pull request merged**